### PR TITLE
Adding Tests for ShareContentValidation

### PR DESCRIPTION
**Pull request by CristianOliveiraDaRosa**
Hi Guys.

I've reviewing the tests when I found this [TODO] (https://github.com/facebook/facebook-android-sdk/blob/master/facebook/tests/src/com/facebook/widget/FacebookDialogTests.java#L86) 

I've added the tests, but this class validates a lot of things. So I am not sure if it covers all the paths.

I've had to refact Validate.class cause this guy has a method named "hasContentProvider" but it was a void method and was hard to mock it and do Unit Test. I changed this to return a boolean result.

Feel free to review.

### CODE REVIEW

File: facebook/src/com/facebook/internal/Validate.java [hunk 1]
```diff
@@ -188,16 +188,21 @@ public static void hasFacebookActivity(Context context, boolean shouldThrow) {
         }
     }
 
-    public static void hasContentProvider(Context context) {
+    public static boolean hasContentProvider(Context context) {
```
Comment (swiese) [inline facebook/src/com/facebook/internal/Validate.java]: Having this not throw does not match the pattern of the other methods in this file nor a typical validate or assert class. 
Is there a reason why you changed this to return a boolean?

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        ShareContentValidation.validateForMessage(linkContent);
+        ShareContentValidation.validateForNativeShare(linkContent);
+        ShareContentValidation.validateForWebShare(linkContent);
+    }
+
+    @Test
+    public void testItAcceptsSharePhotoContent() throws MalformedURLException, URISyntaxException {
+        String AppId = "200";
+
+        PowerMockito.mockStatic(Validate.class);
```
Comment (swiese) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: You don't need to mock the whole class you can just do 
PowerMockito.spy(Validate.class);

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        when(Validate.hasContentProvider(RuntimeEnvironment.application)).thenReturn(true);
```
Comment (swiese) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: Instead of the return true and changing the other classes you can do something along the lines of 
doNothing().when(Validate.class, "hasContentProvider", RuntimeEnvironment.application);

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    public void testItValidatesEmptyListOfPhotoForPhotoShareByMessage() throws MalformedURLException {
+        SharePhotoContent sharePhoto = new SharePhotoContent.Builder().build();
+
+        ShareContentValidation.validateForMessage(sharePhoto);
+    }
+
+    @Test(expected = FacebookException.class)
+    public void testItValidatesMaxSizeOfPhotoShareByMessage() throws MalformedURLException {
+        SharePhotoContent sharePhotoContent =
+                new SharePhotoContent.Builder().addPhoto(buildSharePhoto("https://facebook.com/awesome-1.gif"))
```
Comment (swiese) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: nit put addPhoto on next line

**Comment by CristianOliveiraDaRosa** [comment]
Hi @swiese 
Thanks. I will change as you suggested!

**Comment by CristianOliveiraDaRosa** [comment]
Hey @swiese, the spy solution didn't work. Only mocking with "mockStatic".

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+
+/**
+ * Tests for {@link ShareContentValidation}
+ */
+@RunWith(RobolectricTestRunner.class)
```
Comment (swiese) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: Not needed. Declared on base class

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        ShareContentValidation.validateForMessage(linkContent);
+        ShareContentValidation.validateForNativeShare(linkContent);
+        ShareContentValidation.validateForWebShare(linkContent);
+    }
+
+    @Test
+    public void testItAcceptsSharePhotoContent() throws Exception {
+        String AppId = "200";
+
+        PowerMockito.mockStatic(Validate.class);
```
Comment (swiese) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: This worked fine for me 
        PowerMockito.spy(Validate.class);
        doNothing().when(Validate.class, "hasContentProvider", Mockito.any(Context.class));

File: facebook/src/com/facebook/share/internal/ShareContentValidation.java [hunk 7]
```diff
@@ -49,9 +49,9 @@
  */
 public class ShareContentValidation {
 
-    private static Validator WebShareValidator;
-    private static Validator DefaultValidator;
-    private static Validator ApiValidator;
+    private static Validator webShareValidator;
```
Comment (swiese) [inline facebook/src/com/facebook/share/internal/ShareContentValidation.java]: These changes are unrelated to the diff. Can you revert?

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java (hunk 6 above)
Comment (CristianOliveiraDaRosa) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: You are right. Using Mockito.any(Context.class) worked like a charm.

File: facebook/src/com/facebook/share/internal/ShareContentValidation.java (hunk 7 above)
Comment (CristianOliveiraDaRosa) [inline facebook/src/com/facebook/share/internal/ShareContentValidation.java]: For sure.
I would like to know... Why were these variables named like classes?

File: facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java (hunk 5 above)
Comment (CristianOliveiraDaRosa) [inline facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java]: Done.

**Comment by CristianOliveiraDaRosa** [comment]
Hey @swiese. I did some unnecessaries commits (reverts). Should I squash them?

**Comment by swiese** [comment]
This looks good. Thanks for your contribution!
